internal/api/v1/docs/application.go

```diff
@@ -0,0 +1,242 @@
+package docs
+
+//go:generate swagger generate spec
+
+import "github.com/epinio/epinio/pkg/api/core/v1/models"
+
+// swagger:route GET /applications application AllApps
+// Return list of applications in all namespaces.
+// responses:
+//   200: AppsResponse
+
+// swagger:parameters AllApps
+type AllAppsParam struct{}
+
+// response: See Apps.
+
+// swagger:route GET /namespaces/{namespace}/applications application Apps
+// Return list of applications in the `namespace`.
+// responses:
+//   200: AppsResponse
+
+// swagger:parameters Apps
+type AppsParam struct {
+	Namespace string
+}
+
+// swagger:response AppsResponse
+type AppsResponse struct {
+	// in: body
+	Body models.AppList
+}
+
+// swagger:route POST /namespaces/{namespace}/applications application AppCreate
+// Create the posted new application in the `namespace`.
+// responses:
+//   200: AppCreateResponse
+
+// swagger:parameters AppCreate
+type AppCreateParam struct {
+	Namespace string
+	// in: body
+	Service models.ApplicationCreateRequest
+}
+
+// swagger:response AppCreateResponse
+type AppCreateResponse struct {
+	// in: body
+	Body models.Response
+}
+
+// swagger:route GET /namespaces/{namespace}/applications/{app} application AppShow
+// Return details of the named `application` in the `namespace`.
+// responses:
+//   200: AppShowResponse
+
+// swagger:parameters AppShow
+type AppShowParam struct {
+	Namespace string
+	App       string
+}
+
+// swagger:response AppShowResponse
+type AppShowResponse struct {
+	// in: body
+	Body models.App
+}
+
+// swagger:route GET /namespaces/{namespace}/applications/{app}/logs application AppLogs
+// Return logs of the named `application` in the `namespace`.
+// responses:
+//   200: AppLogsResponse
+
+// swagger:parameters AppLogs
+type AppLogsParam struct {
+	Namespace string
+	App       string
+}
+
+// swagger:response AppLogsResponse
+type AppLogsResponse struct {
+	// in: body
```

Comment:
I haven't found what to add for the `LogsResponse`... Any hints?

Reply:
Heh. The log endpoints upgrade the connection to a websocket and then transfer log entries over it. They do not have a regular response per se I believe.

A very short google does not net me info on how to describe this for swagger. ... If you have more luck let us use that. And if not I would leave the response off, and make note of the websocket upgrade in the endpoint description.

Reply:
Yes, it seems that swagger is not able to describe websocket connection.

So I can change the endpoint description to something like this: `Return logs of the named 'application' in the 'namespace' streamed over a websocket.`
WDYT?

Reply:
I've pushed a comment change, let me know if this is not OK for you.

Reply:
I am :+1: on that change.